### PR TITLE
Add get-pants.sh script.

### PR DESCRIPTION
**Pull request by jsirois**
This will be made available on a rustup.rs-like site and can be
checked-in to a repo to allow new users to on-board easily. The first
cut does not include a Power Shell variant, although it does work with
Git bash for Windows.

Work towards #35

### CODE REVIEW

**Comment by jsirois** [comment]
For example, in setup mode - no Pants, no Pants repo:
```
$ mkdir /tmp/foo
$ cd /tmp/foo
$ /usr/bin/env bash -c "$(curl -fsSL file:///home/jsirois/dev/pantsbuild/jsirois-scie-pants/get-pants.sh)"
Downloading and installing scie-pants release ...
Installed scie-pants 0.2.0 from https://github.com/pantsbuild/scie-pants/releases/latest/download/scie-pants-linux-x86_64 to /home/jsirois/bin/pants
Launching pants ...
No Pants configuration was found at or above /tmp/foo.
Would you like to configure /tmp/foo as a Pants project? (Y/n): y
Fetching latest stable Pants version since none is configured
Creating /tmp/foo/pants.toml and configuring it to use Pants 2.14.0
10:09:13.31 [INFO] Initializing scheduler...
10:09:13.39 [INFO] Scheduler initialized.
10:09:13.40 [WARN] Please either set `enabled = true` in the [anonymous-telemetry] section of pants.toml to enable sending anonymous stats to the Pants project to aid development, or set `enabled = false` to disable it. No telemetry sent for this run. An explicit setting will get rid of this message. See https://www.pantsbuild.org/v2.14/docs/anonymous-telemetry for details.
2.14.0
$ cat pants.toml
[GLOBAL]
pants_version = "2.14.0"
```

File: get-pants.sh [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    if (($# > 0)); then
+      check_cmd rm
+      _GC+=("$@")
+    else
+      rm -rf "${_GC[@]}"
+    fi
+  fi
+}
+
+trap gc EXIT
```
Comment (kaos) [inline get-pants.sh:36]: neat trick 👀

File: get-pants.sh [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  if [[ "${os}" =~ [Ll]inux ]]; then
+    echo linux
+  elif [[ "${os}" =~ [Dd]arwin ]]; then
+    echo macos
+  elif [[ "${os}" =~ [Ww]in|[Mm][Ii][Nn][Gg] ]]; then
+    # Powershell reports something like: Windows_NT
+    # Git bash reports something like: MINGW64_NT-10.0-22621
+    echo windows
+  else
+    die "Integration tests are not supported for this operating system (${os})."
```
Comment (kaos) [inline get-pants.sh:53]: This message gives me pause.. if this script is run by a user on an unsupported OS, and they see "Integration tests are not supported for this operating system ...", well if they want to get up and running with Pants, what has integration tests got to do with it? (I see that may be the reason why it's not available to them, but I find it a not very helpful message to convey)

**Review by kaos — APPROVED** [review]

File: get-pants.sh (hunk 2 above)
Comment (jsirois) [reply get-pants.sh:53]: Copy-pasta: https://github.com/a-scie/jump/blob/main/examples/run.sh
Thanks for catching that.

File: get-pants.sh [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+function calculate_arch() {
+  local arch
+
+  arch="$(uname -m)"
+  if [[ "${arch}" =~ x86[_-]64 ]]; then
+    echo x86_64
+  elif [[ "${arch}" =~ arm64|aarch64 ]]; then
+    echo aarch64
+  else
+    die "Pants is not supported for this chip architecture (${arch})."
```
Comment (jsirois) [inline get-pants.sh:129]: This one I had fixed, just missed the OS message.

Comment (kaos) [reply get-pants.sh:129]: ah, yes noticed this one after writing the other comment... 👍🏽

**Pull request merged**